src/matchmaking/player.hpp

```diff
@@ -30,17 +30,21 @@ class Player {
     /// @brief remove the elapsed time from the participant's time
     /// @param elapsed_millis
     /// @return `false` when out of time
-    [[nodiscard]] bool updateTime(const int64_t elapsed_millis) {
+    [[nodiscard]] bool updateTime(const int64_t elapsed_millis, const int64_t timemargin) {
         if (engine.getConfig().limit.tc.time == 0) {
             return true;
         }
 
         time_control_.time -= elapsed_millis;
 
-        if (time_control_.time < 0) {
+        if (time_control_.time < -timemargin) {
             return false;
         }
 
+        if (time_control_.time < 0) {
+            time_control_.time = 0;
+        }
```

Comment:
i dont really get this, why set it to 0?

Reply:
so it prevents negative time which i feel might lead to weird behavior,
and the way i see it, the timemargin is just an extra time after the actual time control that the engine is alloted, so after the move is made it shouldnt be part of the calculation for the final time after increment is added, 
say an engine exceeded the timelimit by 250ms with a 1 second increment, this way the final time after the move should be 1 second instead of 750ms